src/Deposit.js

```diff
@@ -1099,4 +1103,174 @@ export default class Deposit {
+   * @return {number} Percentage collateralisation, as an integer. eg. 150%
+   */
+  async getInitialCollateralizedPercentage() {
+    return parseInt(
+      await this.contract.methods.getInitialCollateralizedPercent().call()
+    )
+  }
+
+  /**
+   * Get the first threshold for deposit undercollateralisation.
+   * If the collateralization level falls below this percentage, the Deposit can
+   * get courtesy-called.
+   * The deposit can be courtesy called using `Deposit.notifyCourtesyCall`.
+   * @return {number} Percentage collateralisation, as an integer. eg. 125%
+   */
+  async getUndercollateralizedThresholdPercent() {
+    return parseInt(
+      await this.contract.methods
+        .getUndercollateralizedThresholdPercent()
+        .call()
+    )
+  }
+
+  /**
+   * Get the threshold for severe deposit undercollateralisation.
+   * If the collateralization level falls below this percentage, the Deposit
+   * can be liquidated.
+   * Liquidation can be initiated using `Deposit.notifyUndercollateralizedLiquidation`.
+   * @return {number} Percentage collateralisation, as an integer. eg. 110%
+   */
+  async getSeverelyUndercollateralizedThresholdPercent() {
+    return parseInt(
+      await this.contract.methods
+        .getSeverelyUndercollateralizedThresholdPercent()
+        .call()
+    )
+  }
+
+  /**
+   * Notify the contract that the signers are undercollateralized,
+   * and move the deposit into a pre-liquidation state.
+   */
+  async notifyCourtesyCall() {
+    await EthereumHelpers.sendSafely(this.contract.methods.notifyCourtesyCall())
+  }
+
+  /**
+   * Exit the courtesy call state.
+   * Only callable if the deposit is sufficiently collateralised.
+   */
+  async exitCourtesyCall() {
+    await EthereumHelpers.sendSafely(this.contract.methods.exitCourtesyCall())
+  }
+
+  /**
+   * Notify the contract that the deposit is severely undercollateralised,
+   * and begin liquidation of the signer bonds.
+   *
+   * The bonds are auctioned in a falling-price auction. The value of
+   * the bonds can be queried using `Deposit.auctionValue`, and bids placed
+   * using `Deposit.purchaseSignerBondsAtAuction`.
+   */
+  async notifyUndercollateralizedLiquidation() {
+    await EthereumHelpers.sendSafely(
+      this.contract.methods.notifyUndercollateralizedLiquidation()
+    )
+  }
+
+  /**
+   * Purchases the signer bonds and closes the liquidation auction.
+   *
+   * Note: The caller must approve the Deposit contract to spend
+   * `lotSize` amount of TBTC.
+   */
+  async purchaseSignerBondsAtAuction() {
+    await EthereumHelpers.sendSafely(
+      this.contract.methods.purchaseSignerBondsAtAuction()
+    )
+  }
+
+  /**
+   * Gets the current value of signer bonds at auction.
+   * Only callable if the deposit is in the liqudation state.
+   * @return {BN} auction value in wei.
+   */
+  async auctionValue() {
+    return toBN(await this.contract.methods.auctionValue().call())
```

Comment:
We should probably either return numbers or BNs, but do so consistently (or BigNumbers, or whatever).

Reply:
Mmmmm, is there any reason against using number solely for the collateralisation thresholds? It reads sooo much more cleanly.

Reply:
Hmm… How so? My main concern is that it makes our own return values mutually incompatible. So if `getTBTCLotSize` returns a `BN` and you want to apply a `number` collateralization threshold to it (say, for display purposes), you have to convert the `number` to a `BN`. Ditto for `getSatoshiLotSize`. It's more for folks to keep mental track of---am I interacting with a `BN` or a `number`? While I don't love the `BN` API (+ limitations), I like having to track disparate types even less.

And while `collateralisation < thresholdCourtesy` is more readable than `collateralisation.lt(thresholdCourtesy)`, I feel like it's not enough so to make the tradeoff worth it.

Reply:
> So if getTBTCLotSize returns a BN and you want to apply a number collateralization threshold to it (say, for display purposes), you have to convert the number to a BN.

Ah, of course. Yeah, BN-to-go it is.